website/src/components/examples-sandbox.tsx

```diff
@@ -0,0 +1,158 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import { ArrowIcon, CaretSlimIcon, cn } from '@theguild/components';
+
+export const EXAMPLES = {
+  OpenAPI: {
+    'JavaScript Wiki': 'openapi-javascript-wiki',
+    'Location Weather': 'openapi-location-weather',
+    YouTrack: 'openapi-youtrack',
+    Stripe: 'openapi-stripe',
+    StackExchange: 'openapi-stackexchange',
+    'WeatherBit on React App': 'openapi-react-weatherbit',
+    'NextJS with Apollo': 'nextjs-apollo-example',
+  },
+  'JSON Schema': {
+    'Fake API': 'json-schema-example',
+    'Covid-19 Statistics': 'json-schema-covid',
+    'Subscriptions, Webhooks & Live Queries': 'json-schema-subscriptions',
+  },
+  OData: {
+    TripPin: 'odata-trippin',
+    'Microsoft Graph': 'odata-microsoft',
+  },
+  SOAP: {
+    'Country Info': 'soap-country-info',
+  },
+  MySQL: {
+    Rfam: 'mysql-rfam',
+  },
+  SQLite: {
+    Chinook: 'sqlite-chinook',
+  },
+  'Apollo Federation': {
+    'Apollo Federation Example': 'federation-example',
+  },
+  'Apache Thrift': {
+    Calculator: 'thrift-calculator',
+  },
+  gRPC: {
+    'Movies Example': 'grpc-example',
+  },
+  Neo4J: {
+    'Movies Example': 'neo4j-example',
+  },
+};
+
+/**
+ * These options won't work if the Sandbox is a DevBox (not Legacy).
+ * @see https://codesandbox.io/docs/learn/sandboxes/embedding
+ */
+const CODESANDBOX_OPTIONS = {
+  codemirror: '1',
+  hidedevtools: '1',
+  hidenavigation: '1',
+  view: 'editor',
+  module: '%2F.meshrc.yml',
+};
+
+export interface ExamplesSandboxProps extends React.HTMLAttributes<HTMLElement> {
+  preventStealingFocusWithUnpleasantDelay?: boolean;
+  lazy?: boolean;
+}
+
+export function ExamplesSandbox({
+  preventStealingFocusWithUnpleasantDelay,
+  lazy = false,
+  ...rest
+}: ExamplesSandboxProps) {
+  const [exampleDir, setExampleDir] = useState('json-schema-example');
+  const [isVisible, setIsVisible] = useState(lazy ? false : true);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      entries => {
+        entries.forEach(entry => {
+          if (entry.isIntersecting) {
+            setIsVisible(true);
+            observer.disconnect();
+          }
+        });
+      },
+      {
+        rootMargin: '100px',
+        threshold: 0.1,
+      },
+    );
+
+    if (containerRef.current) {
+      observer.observe(containerRef.current);
+    }
```

Comment:
```suggestion
      observer.observe(containerRef.current);
```
you will not get type error